### ISSUE TITLE
`pipenv install envparse` fails

### ISSUE THREAD

**Issue by sobolevn** [kennethreitz/pipenv#792]
I have upgraded `pipenv` to `8.2.5`. Now, I am trying to install `envparse`. And I have faced this issue:
- `pipenv install envparse` fails
- `pipenv uninstall envparse` fails with the same error
- `pipenv lock` fails with the same error (this package was removed already)

As I can see this, package resolution is broken for some reason.

## Output

```
» pipenv install envparse 
Did you mean parse? [Y/n]: n
Installing envparse…
Collecting envparse
  Downloading envparse-0.2.0.tar.gz
Building wheels for collected packages: envparse
  Running setup.py bdist_wheel for envparse: started
  Running setup.py bdist_wheel for envparse: finished with status 'done'
  Stored in directory: /Users/sobolev/Library/Caches/pip/wheels/15/62/ab/22c380e7c640f40265b3ab6d08858d78d125c5dfb89552267d
Successfully built envparse
Installing collected packages: envparse
Successfully installed envparse-0.2.0

Adding envparse to Pipfile's [packages]…
Locking [dev-packages] dependencies…
Traceback (most recent call last):
  File "/usr/local/bin/pipenv", line 11, in <module>
    sys.exit(cli())
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/cli.py", line 1849, in install
    do_lock(system=system, pre=pre)
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/cli.py", line 1024, in do_lock
    pre=pre
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/utils.py", line 421, in resolve_deps
    resolved_tree.update(resolver.resolve())
  File "/Users/sobolev/.pipsi/pipenv/lib/python2.7/site-packages/pipenv/patched/piptools/resolver.py", line 97, in resolve
    'This is likely a bug.' % max_rounds)
RuntimeError: No stable configuration of concrete packages could be found for the given constraints after 12 rounds of resolving.
This is likely a bug.
```

## Pipfile

Here's my `Pipfile`:

```
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true

[requires]
python_version = '3.6'

[dev-packages]

"django-debug-toolbar" = "*"
"pep8-naming" = "*"
"flake8-builtins" = "==0.2"
"flake8-commas" = "==0.4.3"
"flake8-quotes" = "==0.9.0"
"flake8-blind-except" = "*"
pytest = "==3.0.7"
"pytest-flake8" = "==0.8.1"
"pytest-isort" = "==0.1.0"
"pytest-django" = "*"
"pytest-xdist" = "*"
"pytest-cov" = "*"
"pytest-env" = "*"
gitlint = "==0.8.2"
"pre-commit" = "*"
xenon = "*"
radon = "*"
ipython = "==6.2.1"
sphinx = "*"
recommonmark = "*"

[packages]

django = "<1.12"
"django-split-settings" = "*"
"django-axes" = "*"
"psycopg2" = "*"
gunicorn = "*"
envparse = "*"
```

I can not generate `Pipfile.lock`, attached `pipenv lock --verbose` here: https://pastebin.com/spdJgfnU

Thanks!

**Comment by sobolevn** [comment]
Btw, removing `ipython` and `Sphinx` solved the issue. 
But that's definitely not what I want.

**Comment by kennethreitz** [comment]
Cannot reproduce.

**Comment by kennethreitz** [comment]
`pipenv lock --verbose` will shed some light on the issue. we can try raising the number of rounds

**Comment by kennethreitz** [comment]
raised the number of rounds

**Comment by sobolevn** [comment]
Is it possible to configure the number of round? I guess, it could be a nice feature. 

With sane defaults.

**Comment by kennethreitz** [comment]
I just did that :)

`PIPENV_MAX_ROUNDS`. Will be in the next release.

**Comment by sobolevn** [comment]
@kennethreitz still no luck. even tried with `PIPENV_MAX_ROUNDS=100`.

**Comment by vphilippon** [comment]
I've noticed that in the resolving part, some packages have a version pinning added and removed on pretty much every round, never stabilising:
This kind of thing:
```
New dependencies found in this round:
  adding [u'appnope', '', '[]']
  adding [u'pathlib2', '', '[]']
  adding [u'pexpect', '==4.2.1', '[]']
  adding [u'typing', '==3.6.2', '[]']
Removed dependencies in this round:
  removing [u'appnope', '==0.1.0', '[]']
  removing [u'pathlib2', '==2.3.0', '[]']
  removing [u'pexpect', '', '[]']
  removing [u'typing', '', '[]']
```
I've spotted that `appnope==0.1.0` is added, while no packages seems to clearly require `==0.1.0`.
From the `pip-tools` resolving perspective, I don't know why that pin is added.
Oddly, the "unstable dependencies" seems to have the "from line" (`appnope==0.1.0 from https://pypi.python.org/packages/87/a9/7985e6a53402f294c8f0e8eff3151a83f1fb901fa92909bb3ff29b4d22af/appnope-0.1.0-py2.py3-none-any.whl#md5=a73064be99839b9534b4da47a16c9c46
`), which is usually dropped by `pip-tools` resolving.

Maybe this will help figuring out what's going on.
